yew-functional-macro/tests/functional_macro/no-name-fail.rs

```diff
@@ -0,0 +1,19 @@
+use yew::prelude::*;
+use yew_functional::{functional_component};
+
+#[derive(Clone, Properties, PartialEq)]
+struct Props {
+    a: usize,
+}
+
+#[functional_component()]
```

Comment:
The error message here should be improved by my suggestion above. Could you also add a test for the following cases:
- invalid Rust ident
- Rust keyword
- a case where there are tokens following the ident: `#[functional_component(a, b, c)]` (I believe this currently doesn't fail and just silently ignores the extra tokens which probably isn't what we want)
- what happens when the component name is the same as the function?

Reply:
> (I believe this currently doesn't fail and just silently ignores the extra tokens which probably isn't what we want)

It does fail with `error: unexpected token`. I will add those.